agent/data_item.cpp

```diff
@@ -78,9 +78,18 @@ DataItem::DataItem(std::map<string, string> attributes)	:
 	m_conversionRequired(false),
 	m_hasFactor(false)
 {
-	m_id = attributes["id"];
-	m_name = attributes["name"];
-	m_type = attributes["type"];
+	auto const idPos = attributes.find("id");
```

Comment:
Although these changes do not look stylistically as simple as the lines they are replacing, they are more efficient. If an item is not found in the map then the old code would have created a new key-value object in the map and then returned its default value. This pollutes the map creating entries when it should not have had any.

This meant client code was always checking for existence and the default 'empty/null' value. By using this style the value does not have to be subsequently checked for 'empty/null'.

When constructing a 'DataItem' a temporary map object is no longer created, which is not much effort for modern CPUs but every little counts.